pkg/provider/azure_vmss_cache.go

```diff
@@ -83,6 +89,23 @@ func (ss *ScaleSet) newVMSSCache() (*azcache.TimedCache, error) {
 			}
 		}
 
+		if resourceGroupNotFound {
+			// gc vmss vm cache when there is resource group not found
+			removed := map[string]bool{}
+			ss.vmssVMCache.Range(func(key, value interface{}) bool {
```

Comment:
Why not just empty all records in cache? Like ss.vmssVMCache = newVmssVMCache().

Reply:
the records should be removed in a safe way. Creating a new data structure would have potential race conditions when the code is invoked from multiple controllers.